core/src/com/unciv/logic/automation/civilization/NextTurnAutomation.kt

```diff
@@ -514,23 +514,33 @@ object NextTurnAutomation {
     }
 
     private fun chooseTechToResearch(civInfo: Civilization) {
-        if (civInfo.tech.techsToResearch.isEmpty()) {
+        fun getGroupedResearchableTechs(): List<List<Technology>> {
             val researchableTechs = civInfo.gameInfo.ruleset.technologies.values
-                    .filter { civInfo.tech.canBeResearched(it.name) }
-            val techsGroups = researchableTechs.groupBy { it.cost }
-            val costs = techsGroups.keys.sorted()
-
-            if (researchableTechs.isEmpty()) return
+                .asSequence()
+                .filter { civInfo.tech.canBeResearched(it.name) }
+                .groupBy { it.cost }
+            return researchableTechs.toSortedMap().values.toList()
+        }
+        while(civInfo.tech.freeTechs > 0) {
+            val costs = getGroupedResearchableTechs()
+            if (costs.isEmpty()) return
 
-            val cheapestTechs = techsGroups[costs[0]]!!
+            val mostExpensiveTechs = costs[costs.size - 1]
+            civInfo.tech.getFreeTechnology(mostExpensiveTechs.random().name)
+        }
+        if (civInfo.tech.techsToResearch.isEmpty()) {
+            val costs = getGroupedResearchableTechs()
+            if (costs.isEmpty()) return
+            
+            val cheapestTechs = costs[0]
             //Do not consider advanced techs if only one tech left in cheapest group
             val techToResearch: Technology =
                 if (cheapestTechs.size == 1 || costs.size == 1) {
                     cheapestTechs.random()
                 } else {
                     //Choose randomly between cheapest and second cheapest group
-                    val techsAdvanced = techsGroups[costs[1]]!!
-                    (cheapestTechs + techsAdvanced).random()
+                    val techsAdvanced = costs[1]
+                    techsAdvanced.toCollection(cheapestTechs.toMutableList()).random()
```

Comment:
This is a little confusing as a way to add 2 lists together - if you can find a clearer way?
I think that "+" works on lists, so just (a + b).random()

Reply:
That's what was originally there but SomeTroglodyte recommended using toMutableSet() before I switched it to a list.
> Oh my - that `+` in existing code operates on Lists - never optimal...
I don't know enough about how Kotlin implements these collections to know which one is better. The collections will be really small, meaning performance isn't much of an issue, so I think the + operation looks better.

Reply:
😓 
Personally I prefer readability where performance is not an issue, and I feel this is one of those cases, but I'm good with either, your choice